### PR TITLE
Bug 1853734: Reusing Azure API key

### PR DESCRIPTION
**Pull request by ricardomaraschini**
Reusing the same primary access key, caching it for five minutes.

### CODE REVIEW

File: pkg/storage/azure/cached_key.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		metrics.AzureKeyCacheHit()
+		return k.value, nil
+	}
+
+	keysResponse, err := cli.ListKeys(ctx, resourceGroup, account)
+	if err != nil {
+		return "", err
+	}
+
+	metrics.AzureKeyCacheMiss()
```
Comment (dmage) [inline pkg/storage/azure/cached_key.go:43]: it should be before ListKeys

File: pkg/storage/azure/azure_test.go [hunk 2]
```diff
@@ -722,6 +722,7 @@ func Test_assureContainer(t *testing.T) {
 			drv := NewDriver(context.Background(), storageConfig, listers)
 			drv.authorizer = autorest.NullAuthorizer{}
 			drv.sender = sender
+			primaryKey = cachedKey{}
```
Comment (dmage) [inline pkg/storage/azure/azure_test.go:725]: is it in order to clear cache after another test?

File: pkg/storage/azure/cached_key.go (hunk 1 above)
Comment (ricardomaraschini) [reply pkg/storage/azure/cached_key.go]: right.

File: pkg/storage/azure/azure_test.go (hunk 2 above)
Comment (ricardomaraschini) [reply pkg/storage/azure/azure_test.go:725]: It does not change much, with this in place we can avoid having different mocked http responses when the key is cached. It gives all tests on that test table the same environment.

**Pull request merged**